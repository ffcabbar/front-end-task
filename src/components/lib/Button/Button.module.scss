.button {
  background: #12b8ff;
  color: #fff;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.05);
  border-radius: 2px;
  border: none;
  margin: 0;
  width: auto;
  overflow: visible;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  padding: 0.5rem 1rem;
  cursor: pointer;
  transition-duration: 0.2s;
  transition-timing-function: cubic-bezier(0.2, 0.8, 0.4, 1);

  &:hover {
    background: darken(#12b8ff, 5%);
    transform: translateY(-2px);
  }
}
