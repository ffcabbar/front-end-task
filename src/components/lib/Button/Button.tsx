import styles from './Button.module.scss';

type Props = {
  children?: React.ReactNode;
};

export const Button = ({ children }: Props) => {
  return <button className={styles.button}>{children}</button>;
};
