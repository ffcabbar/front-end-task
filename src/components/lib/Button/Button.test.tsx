import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Button } from './Button';

test('renders button', () => {
  render(<Button children="Kuehne Nagel" />);
  const buttonElement = screen.getByText(/Kuehne Nagel/i);
  expect(buttonElement).toBeInTheDocument();
});

test('handles onClick', () => {
  const onClick = jest.fn();
  render(<Button onClick={onClick} children="Kuehne Nagel" />);
  const buttonElement = screen.getByText('Kuehne Nagel');
  fireEvent.click(buttonElement);
  expect(onClick).toHaveBeenCalledTimes(1);
});

test('matches snapshot', () => {
  const tree = renderer.create(<Button children="Kuehne Nagel" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches snapshot correctly when there are no props', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
