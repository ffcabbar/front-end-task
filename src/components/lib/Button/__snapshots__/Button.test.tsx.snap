// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="button"
>
  Kuehne Nagel
</button>
`;

exports[`matches snapshot correctly when there are no props 1`] = `
<button
  className="button"
/>
`;
